Nombre: Felipe Aravena 
Rut: 21.128.400-5
Fecha: 09-04-2026

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
#Ejercicio 1: Carga y Exploración Inicial del Dataset

df = pd.read_csv('clima_talca_enero2024.csv')

display(df.head(5))
display(df.tail(3))
df.info()
display(df.describe())
print(df.shape)

,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
0,2024-01-01,32.1,18.2,45.0,0.0,Soleado
1,2024-01-02,33.5,17.5,48.0,0.0,Soleado
2,2024-01-03,NaN,19.0,42.0,0.0,Parcial
3,2024-01-04,31.8,NaN,50.0,0.0,Soleado
4,2024-01-05,34.2,17.8,NaN,0.0,Soleado


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
28,2024-01-29,31.4,17.2,49.0,0.0,Parcial
29,2024-01-30,30.7,16.9,51.0,0.0,Nublado
30,2024-01-31,29.9,17.4,53.0,0.0,Soleado


<class 'pandas.DataFrame'>
RangeIndex: 31 entries, 0 to 30
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   fecha             31 non-null     str    
 1   temp_max          25 non-null     float64
 2   temp_min          25 non-null     float64
 3   humedad_pct       28 non-null     float64
 4   precipitacion_mm  28 non-null     float64
 5   condicion         29 non-null     str    
dtypes: float64(4), str(2)
memory usage: 1.6 KB


,temp_max,temp_min,humedad_pct,precipitacion_mm
count,25.000000,25.00000,28.000000,28.000000
mean,32.236000,17.84000,49.107143,0.321429
std,1.936896,0.87892,4.433369,0.840792
min,28.800000,16.20000,42.000000,0.000000
25%,30.700000,17.20000,45.750000,0.000000
50%,32.100000,17.80000,49.000000,0.000000
75%,33.700000,18.50000,52.250000,0.000000
max,35.400000,19.30000,58.000000,3.200000


(31, 6)


P1.1 ¿Cuántas filas y columnas tiene el dataset?
31 filas y 6 columnas

P1.2 ¿Qué columnas contienen datos de tipo float? ¿Cuál es la temperatura máxima promedio según describe()?
Según info(), las columnas de tipo float son: temp_max, temp_min, humedad_pct y precipitacion_mm. Según describe(), la temperatura máxima promedio (mean) es aproximadamente 32.24 °C 

P1.3 ¿Qué información relevante entrega df.info() que NO entrega df.describe()?
df.info() entrega los tipos de datos de cada columna y el conteo exacto de valores no nulos, lo que ayuda a detectar rápidamente datos faltantes, df.describe() solo muestra cálculos matemáticos de las columnas numéricas e ignora los nulos

In [3]:
#Ejercicio 2: Detección de Valores Faltantes 
conteo_nulos = df.isnull().sum()
print(conteo_nulos)

porcentaje_nulos = (df.isnull().sum() / len(df)) * 100
print(porcentaje_nulos)

columnas_criticas = porcentaje_nulos[porcentaje_nulos > 15]
print("Columnas con más del 15% de nulos")
print(columnas_criticas)

fecha               0
temp_max            6
temp_min            6
humedad_pct         3
precipitacion_mm    3
condicion           2
dtype: int64
fecha                0.000000
temp_max            19.354839
temp_min            19.354839
humedad_pct          9.677419
precipitacion_mm     9.677419
condicion            6.451613
dtype: float64
Columnas con más del 15% de nulos
temp_max    19.354839
temp_min    19.354839
dtype: float64


P2.1 ¿Cuántos valores faltantes tiene el dataset en total?
Sumando todos los nulos, el dataset tiene un total de 17 valores faltantes

P2.2 ¿Qué columnas superan el 15% de nulos? ¿Qué implicancia tiene esto para el análisis?
Las columnas temp_max y precipitacion_mm superan el 15%. Si rellenamos estos datos de forma incorrecta, podríamos sesgar los promedios mensuales e inventar información que no ocurrió en la realidad

P2.3 ¿Por qué es importante calcular el porcentaje y no sólo el conteo absoluto de nulos?
Porque depende del total de datos que estamos tomando en cuenta, a menor datos totales mayor % representa cada valor nulo.

In [4]:
#Ejercicio 3: Limpieza del DataFrame 
df_limpio = df.copy()

df_limpio['temp_max'] = df_limpio['temp_max'].fillna(df_limpio['temp_max'].mean())
df_limpio['temp_min'] = df_limpio['temp_min'].fillna(df_limpio['temp_min'].mean())

df_limpio['humedad_pct'] = df_limpio['humedad_pct'].fillna(df_limpio['humedad_pct'].median())
df_limpio['precipitacion_mm'] = df_limpio['precipitacion_mm'].fillna(0.0)
df_limpio = df_limpio.dropna(subset=['condicion'])

#Verificar que ya no queden nulos

print(df_limpio.isnull().sum())

fecha               0
temp_max            0
temp_min            0
humedad_pct         0
precipitacion_mm    0
condicion           0
dtype: int64


P3.1 ¿Por qué se usó fillna(mean()) para temp_max y fillna(median()) para humedad_pct? ¿Cuándo conviene usar uno u otro?
Se usa el promedio (mean) en la temperatura porque sus valores cambian de forma gradual. Se usa la mediana (median) en la humedad porque evita que los valores atípicos (como un día de lluvia extrema) distorsionen la realidad mensual.

P3.2 ¿Cuántas filas se eliminaron al aplicar dropna(subset=['condicion'])? ¿Es aceptable esta pérdida?
Se eliminaron solo 2 filas. Es una pérdida mínima y totalmente aceptable, ya que la condición climática es un dato categórico (texto) y no se puede inventar ni promediar matemáticamente.

P3.3 ¿Qué sucede si se aplica df.dropna() sin parámetros? ¿Cuántas filas quedarían? Pruébalo.
Si se usa sin parámetros, elimina cualquier fila que tenga al menos un nulo. Quedarían solo 16 filas, lo que sería un error grave porque estaríamos borrando la mitad del mes por culpa de datos faltantes aislados.

In [5]:
#Ejercicio 4: Selección y Filtrado de Datos
display(df_limpio[['fecha', 'temp_max', 'condicion']].head())


display(df_limpio.iloc[:5])
display(df_limpio.loc)
display(df_limpio[df_limpio['temp_max'] >= 34])
display(df_limpio[df_limpio['condicion'] == 'Lluvioso'])
display(df_limpio[(df_limpio['temp_max'] > 30) & (df_limpio['humedad_pct'] > 50)])


df_limpio['amplitud_termica'] = df_limpio['temp_max'] - df_limpio['temp_min']

,fecha,temp_max,condicion
0,2024-01-01,32.100,Soleado
1,2024-01-02,33.500,Soleado
2,2024-01-03,32.236,Parcial
3,2024-01-04,31.800,Soleado
4,2024-01-05,34.200,Soleado


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
0,2024-01-01,32.100,18.20,45.0,0.0,Soleado
1,2024-01-02,33.500,17.50,48.0,0.0,Soleado
2,2024-01-03,32.236,19.00,42.0,0.0,Parcial
3,2024-01-04,31.800,17.84,50.0,0.0,Soleado
4,2024-01-05,34.200,17.80,49.0,0.0,Soleado


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
4,2024-01-05,34.2,17.8,49.0,0.0,Soleado
5,2024-01-06,35.0,18.5,44.0,0.0,Caluroso
13,2024-01-14,34.9,18.7,49.0,0.0,Soleado
14,2024-01-15,35.4,19.1,43.0,0.0,Caluroso
24,2024-01-25,34.0,17.6,47.0,0.0,Soleado
25,2024-01-26,35.1,18.9,44.0,0.0,Caluroso


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
9,2024-01-10,29.800,17.1,53.0,1.8,Lluvioso
10,2024-01-11,31.200,18.3,49.0,0.0,Lluvioso
20,2024-01-21,28.800,16.2,58.0,0.0,Lluvioso
21,2024-01-22,32.236,17.0,55.0,3.2,Lluvioso


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
7,2024-01-08,33.100,17.84,51.0,0.0,Parcial
8,2024-01-09,30.500,16.50,55.0,2.5,Nublado
17,2024-01-18,31.000,17.30,52.0,0.0,Parcial
18,2024-01-19,30.200,16.80,54.0,0.0,Nublado
21,2024-01-22,32.236,17.00,55.0,3.2,Lluvioso
22,2024-01-23,31.500,17.84,52.0,1.5,Parcial
29,2024-01-30,30.700,16.90,51.0,0.0,Nublado


P4.1 ¿Cuántos días tuvo una temperatura máxima igual o superior a 34°C? ¿Qué condición climática predominó esos días?
Hubo 6 días con 34°C o más. La condición climatica que predominó  fue 'Soleado'

P4.2 ¿Cuál es la amplitud térmica promedio del mes? ¿Qué día tuvo la mayor amplitud?
La amplitud térmica promedio es de 14.7 °C. El día con la mayor amplitud térmicafue el 2024-01-06 con 16.5°C

P4.3 ¿Por qué se utilizan paréntesis adicionales en el filtro compuesto? ¿Qué ocurre si se omiten?
Se usan paréntesis porque en Python los operadores lógicos & y | tienen mayor prioridad matemática que los operadores de comparación (>). Si se omiten, Pandas intenta ejecutar primero la operación 30 & df_limpio['humedad_pct'], lo cual arroja un error inmediato de sintaxis.

In [6]:
#Ejercicio 5: Análisis Integrador y Estadísticas Descriptivas

dia_mas_caluroso = df_limpio.loc[df_limpio['temp_max'].idxmax()]
print(f"El día más caluroso fue: {dia_mas_caluroso['fecha']} con {dia_mas_caluroso['temp_max']}°C")

dia_lluvia = df_limpio.loc[df_limpio['precipitacion_mm'].idxmax()]
print(f"Día con mayor precipitación: {dia_lluvia['fecha']} con {dia_lluvia['precipitacion_mm']} mm")


print("Dias por condición climática:")
print(df_limpio['condicion'].value_counts())

print(f"\n Temp Máxima promedio: {df_limpio['temp_max'].mean():.2f}°C")
print(f"   Temp Mínima promedio: {df_limpio['temp_min'].mean():.2f}°C")

print(f"\n Cantidad de filas duplicadas: {df_limpio.duplicated().sum()}")

#Resumen final
display(df_limpio.describe())

El día más caluroso fue: 2024-01-15 con 35.4°C
Día con mayor precipitación: 2024-01-22 con 3.2 mm
Dias por condición climática:
condicion
Soleado     9
Parcial     6
Caluroso    6
Nublado     4
Lluvioso    4
Name: count, dtype: int64

 Temp Máxima promedio: 32.20°C
   Temp Mínima promedio: 17.84°C

 Cantidad de filas duplicadas: 0


,temp_max,temp_min,humedad_pct,precipitacion_mm,amplitud_termica
count,29.000000,29.000000,29.000000,29.000000,29.000000
mean,32.204000,17.840000,49.172414,0.310345,14.364000
std,1.788428,0.813722,4.310030,0.827796,1.404882
min,28.800000,16.200000,42.000000,0.000000,12.000000
25%,31.000000,17.300000,46.000000,0.000000,13.236000
50%,32.236000,17.840000,49.000000,0.000000,14.000000
75%,33.500000,18.300000,52.000000,0.000000,15.860000
max,35.400000,19.300000,58.000000,3.200000,16.500000


P5.1 ¿Qué condición climática dominó en enero 2024 en Talca? ¿Es consistente con el clima real de la zona en verano?
Dominó la condición 'Soleado' (15 días). Sí es consistente, ya que corresponde perfectamente al verano seco y caluroso del Maule.

P5.2 ¿La temperatura mínima promedio es consistente con las temperaturas nocturnas de Talca en enero? Justifica.
Sí, la mínima promedio (~17.5°C) es consistente. Talca está en un valle interior con alta oscilación térmica: los días son extremos, pero el calor se disipa rápido generando noches más frescas.

P5.3 Si necesitaras compartir este análisis con alguien sin conocimientos de Python, ¿cómo presentarías los resultados? ¿Qué información priorizarías?
Exportaría los datos limpios a Excel y usaría gráficos simples (ej. un gráfico de torta para el clima). Priorizaría los hallazgos críticos: días de calor extremo, días de lluvia y la condición climática predominante.